src/transformers/models/zamba2/modular_zamba2.py

```diff
@@ -0,0 +1,1872 @@
+                                    + str(adapter_id)
+                                    + ".0.weight"
+                                )
+                                tied_keys_adapter.append(
+                                    "shared_transformer.feed_forward.gate_up_proj_adapter_list."
+                                    + str(adapter_id)
+                                    + ".1.weight"
+                                )
+                            adapter_id += 1
+                        self._tied_weights_keys = [*self._tied_weights_keys, *tied_keys_adapter]
+                    if config.use_shared_attention_adapter:
+                        tied_keys_adapter = []
+                        adapter_id = 0
+                        for _layer_type in self.layers_block_type:
+                            if _layer_type == "hybrid" and adapter_id % config.num_mem_blocks == block.block_id:
+                                tied_keys_adapter.append(
+                                    "shared_transformer.self_attn.linear_q_adapter_list."
+                                    + str(adapter_id)
+                                    + ".0.weight"
+                                )
+                                tied_keys_adapter.append(
+                                    "shared_transformer.self_attn.linear_k_adapter_list."
+                                    + str(adapter_id)
+                                    + ".0.weight"
+                                )
+                                tied_keys_adapter.append(
+                                    "shared_transformer.self_attn.linear_v_adapter_list."
+                                    + str(adapter_id)
+                                    + ".0.weight"
+                                )
+                                tied_keys_adapter.append(
+                                    "shared_transformer.self_attn.linear_q_adapter_list."
+                                    + str(adapter_id)
+                                    + ".1.weight"
+                                )
+                                tied_keys_adapter.append(
+                                    "shared_transformer.self_attn.linear_k_adapter_list."
+                                    + str(adapter_id)
+                                    + ".1.weight"
+                                )
+                                tied_keys_adapter.append(
+                                    "shared_transformer.self_attn.linear_v_adapter_list."
+                                    + str(adapter_id)
+                                    + ".1.weight"
+                                )
+                            adapter_id += 1
+                        self._tied_weights_keys = [*self._tied_weights_keys, *tied_keys_adapter]
+                layers.append(Zamba2HybridLayer(block, next(linear_layers), next(mamba_layers)))
+            else:
+                layers.append(next(mamba_layers))
```

Comment:
Putting this block in another method such as `get_layers` would help lisibility a great deal!

Reply:
Done [here](https://github.com/huggingface/transformers/pull/34517/commits/095d853bf872e9d05036dddf4fd8f011cb8646dd).